x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -0,0 +1,164 @@
+
+    public async assertRowExists(fieldName: string) {
+      await testSubjects.existOrFail(this.rowSelector(fieldName));
+    }
+
+    public async expandRowDetails(fieldName: string, fieldType: string) {
+      const selector = this.rowSelector(
+        fieldName,
+        `mlDataVisualizerToggleDetails ${fieldName} arrowDown`
+      );
+      await testSubjects.existOrFail(selector);
+      await testSubjects.click(selector);
+      await retry.tryForTime(5000, async () => {
+        await testSubjects.existOrFail(
+          this.rowSelector(fieldName, `mlDataVisualizerToggleDetails ${fieldName} arrowUp`)
+        );
+        await testSubjects.existOrFail(
+          `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType}`
+        );
+      });
+    }
+
+    public async assertFieldDocCount(fieldName: string, docCountFormatted: string) {
+      const docCountFormattedSelector = this.rowSelector(
+        fieldName,
+        'mlDataVisualizerTableColumnDocumentsCount'
+      );
+      await testSubjects.existOrFail(docCountFormattedSelector);
+      const docCount = await testSubjects.getVisibleText(docCountFormattedSelector);
+      expect(docCount).to.eql(
+        docCountFormatted,
+        `Expected total document count to be '${docCountFormatted}' (got '${docCount}')`
+      );
+    }
+
+    public async assertNumberRowContents(fieldName: string, docCountFormatted: string) {
+      const fieldType = ML_JOB_FIELD_TYPES.NUMBER;
+      await this.assertRowExists(fieldName);
+      await this.assertFieldDocCount(fieldName, docCountFormatted);
+
+      await this.expandRowDetails(fieldName, fieldType);
+
+      await testSubjects.existOrFail(
+        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlNumberSummaryTable`
+      );
+
+      await testSubjects.existOrFail(
+        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlTopValues`
+      );
+      await testSubjects.existOrFail(
+        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlMetricDistribution`
+      );
+    }
+
+    public async assertDateRowContents(fieldName: string, docCountFormatted: string) {
+      const fieldType = ML_JOB_FIELD_TYPES.DATE;
+      await this.assertRowExists(fieldName);
+      await this.assertFieldDocCount(fieldName, docCountFormatted);
+
+      await this.expandRowDetails(fieldName, fieldType);
+
+      await testSubjects.existOrFail(
+        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlDateSummaryTable`
+      );
+    }
+
+    public async assertKeywordRowContents(fieldName: string, docCountFormatted: string) {
+      const fieldType = ML_JOB_FIELD_TYPES.KEYWORD;
+      await this.assertRowExists(fieldName);
+      await this.assertFieldDocCount(fieldName, docCountFormatted);
+
+      await this.expandRowDetails(fieldName, fieldType);
+
+      await testSubjects.existOrFail(
+        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlFieldDataCardTopValues`
```

Comment:
I'd suggest to remove `Card` from this data test subject as it's not longer content of a card.